src/Stratis.Bitcoin/Base/Deployments/ThresholdConditionCache.cs

```diff
@@ -58,6 +59,49 @@ public ThresholdState[] GetStates(ChainedHeader pindexPrev)
             return array;
         }
 
+        public object GetThresholdStateMetrics(ChainedHeader indexPrev, ThresholdState[] thresholdStates)
+        {
+            var nonNullDeployments = new List<int>();
+            for (int i = 0; i < this.consensus.BIP9Deployments.Length; i++)
```

Comment:
It feels a bit wasteful to keep the deployment in an array and then in a new loop, loop around all the deployments and skip over the null ones.
Any chance we can have one loop?

Reply:
Logic refined as suggested.